### ISSUE TITLE
AIVehicleExtension for FS22?

### ISSUE THREAD

**Issue by rasc1974** [Mogli12/AIVehicleExtension#251]
Hi,
is this mod coming for FS22, as well? I would very much appreciate, because I really like the funtion to work in straight lines as the ingame helper but to add a headland, when doing field work in FS19! I used it in every savegame in FS19.
Best regards
Rainer

**Comment by Mogli12** [comment]
Hi Rainer,
Currently, I am busy with conversion of vehicle control addon. 
Best regards, 
Stefan

**Comment by Clouseau219** [comment]
Hi Stefan,

thx for your work. Looking forward to use your mods.

**Comment by rasc1974** [comment]
Super! Thanks for the info.

**Comment by kf6shc** [comment]
would you be willing to bring on another person to convert AIVE to 22?

**Comment by Mogli12** [comment]
AIVE did not really work in all cases in FS19. I am thinking of something different. At start of a field job AIVE can calculate the compleete path. Subsequently, the vehicle "just" follows this path. The path needs to be stored in the savegame.

My estiamte: it will take weeks or month to implement it.

**Comment by Williwillswisse** [comment]
Hi Stefan, I'm a great fan and "power user" of your mods 😀
As a first step, I think it would be nice if you convert AIVE "as it is" (if it is possible). That would help us a lot. Further developments could follow later.
Thanks a lot for your quick conversion of VCA, I love it.

**Comment by LucasRMartins** [comment]
Hi Stefan, I love you and your mod. I use AIVE for years, and i cant wait for FS22 version. Huge thanks from Brazil.

**Comment by dherschberger** [comment]
I really like this mod as well, I used it a lot in 17 but didn't play 19 very much. I agree with @Williwillswisse that it would be great to convert as is first then do improvements. I like that it didn't compute a path, that worked better so I could put multiple combines in the same field and they knew what to do. If they just follow a computed path, it would just be Courseplay. Thanks!

**Comment by Richy80** [comment]
Ich habe diese Mod im FS19 (ich glaub im FS15 auch) gespielt und habe diesen geliebt ich hatte nie Probleme mit Drescher und bin immer selbst abgefahren mit dem Hänger hat super funktioniert.
Mir wäre sehr geholfen wenn die letzte Version von AIVE im FS22 funktionieren würde.
Mit Courseplay hab ich sie auch kombiniert.

**Comment by Mogli12** [comment]
Eine einfache Konvertierung klappt so noch nicht. Es wirft nur LUA Fehler.

**Comment by Richy80** [comment]
Ahh Du machst das schon ich glaub an Dich... :-)

**Comment by fudoido** [comment]
> AIVE did not really work in all cases in FS19. I am thinking of something different. At start of a field job AIVE can calculate the compleete path. Subsequently, the vehicle "just" follows this path. The path needs to be stored in the savegame.
> 
> My estiamte: it will take weeks or month to implement it.

dont want to make comparisons, but seems that what you want to implement is simmilar to courseplay fieldwork calculation tool, only you are trying to do it automatically when we are in the edge of the field we want to work in. am i correct?
that would be great to be fair. i used AIVE quite alot of times, but sometimes i would get mad with some fails it had, and not sure if it was due the map itself (or the fields), or the machinery i chose for the work, but it would from time to time just starting doing circles around the same spot until it would eventually stop.
if you plan to implement something like what i said above, may i suggest to try calculate the headlands first as much as it can work for them, then doing up/down rows just to finish off the field? or if it can somehow do all the work around the fields, even better i think. 
since i came up with courseplay, maybe you could talk to them and find if you can work with part of the fieldwork generation course coding and use it for your mod, and both can cooperate and maybe they also implement a fully working automatic generation tool. autodrive and courseplay were working together before in fs19, maybe you can work with them also. great minds think better together than solo ;)

**Comment by Williwillswisse** [comment]
Danke Stefan 😃👍

**Comment by ph1go** [comment]
Thank you, Stefan. It's difficult to imagine fieldwork - especially harvesting - without this mod that enables workers to do more than just robotically go up and down fields. I haven't tested it properly yet but it's great to have one of the most useful mods already out for FS22.

**Comment by Mogli12** [comment]
@ph1go Yes, this is just a simple conversion to FS22. I disabled some features. I did not yet decide what to do with AIVE.

**Comment by Richy80** [comment]
DANKE Stefan hast mir sehr toll geholfen, bist der BESTE!!!
